src/main/kotlin/coverosR3z/persistence/types/ChangeTrackingSet.kt

```diff
@@ -65,6 +65,24 @@ class ChangeTrackingSet<T> : ConcurrentSet<T>() {
         return super.remove(element)
     }
 
+    override fun equals(other: Any?): Boolean {
```

Comment:
more explicit with equals, we only want to concern ourselves with relevant differences between similar items.  Before, we were potentially impacted by marginally important behind-the-scenes state differences